sound/soc/sof/intel/hda-dsp.c

```diff
@@ -696,12 +726,42 @@ int hda_dsp_resume(struct snd_sof_dev *sdev)
 		.state = SOF_DSP_PM_D0,
 		.substate = SOF_HDA_DSP_PM_D0I0,
 	};
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct hdac_ext_link *hlink = NULL;
+#endif
 	int ret;
 
 	/* resume from D0I3 */
 	if (sdev->dsp_power_state.state == SOF_DSP_PM_D0) {
 		hda_codec_i915_display_power(sdev, true);
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+		/* power up links that were active before suspend */
+		list_for_each_entry(hlink, &bus->hlink_list, list) {
+			if (hlink->ref_count)
+				ret = snd_hdac_ext_bus_link_power_up(hlink);
+			if (ret < 0) {
```

Comment:
@kv2019i I have an alternate proposal to call the platform trigger from the component driver (pcm.c) instead. Unfortunately, I cant test it out yet because apparently S0ix with WoV is still broken on the TGL board (even with this PR). Will update soon

Reply:
@ranj063 its not WoV what is broken by platform setup (coreboot in this case)
When it comes to your alternate version which hopes to resume and suspend from pcm.c, as I said during recent meeting - suspending this way may work but resume not. As far as I know we don't resume in WoV case by a pcm event. The idea to suspend/resume from pcm.c is not a news for me, in fact I have also tried it a the very beginning as it is natural first choice.